### ISSUE TITLE
Check failed: proto.SerializeToOstream(&output)

### ISSUE THREAD

**Issue by joyofdata** [NVIDIA/DIGITS#84]
```
ERROR: error code -6

Iteration 8100, loss = 0.433653
Train net output #0: loss = 0.433653 (* 1 = 0.433653 loss)
Iteration 8100, lr = 0.001
Iteration 8150, loss = 0.428066
Train net output #0: loss = 0.428066 (* 1 = 0.428066 loss)
Iteration 8150, lr = 0.001
Iteration 8200, loss = 0.431769
Train net output #0: loss = 0.431769 (* 1 = 0.431769 loss)
Iteration 8200, lr = 0.001
Iteration 8250, loss = 0.519064
Train net output #0: loss = 0.519064 (* 1 = 0.519064 loss)
Iteration 8250, lr = 0.001
Iteration 8300, loss = 0.477628
Train net output #0: loss = 0.477628 (* 1 = 0.477628 loss)
Iteration 8300, lr = 0.001
Iteration 8350, loss = 0.482783
Train net output #0: loss = 0.482783 (* 1 = 0.482783 loss)
Iteration 8350, lr = 0.001
Snapshotting to snapshot_iter_8359.caffemodel
Check failed: proto.SerializeToOstream(&output)
```

# Dataset:
```
run_0

Done 02:28:23 PM

Image Size
256x256
Image Type
COLOR
Create DB (train)
64234 images
Create DB (val)
4000 images
Create DB (test)
2000 images
Encoding
png
```

# Logs:

- [solver.prototxt](http://pastebin.com/YgRmek7R)
- [train_val.prototxt](http://pastebin.com/ihf57cwG)
- [deploy.prototxt](http://pastebin.com/tBfuf9te)
- [caffe_output.log](http://pastebin.com/Kp3Cqz6k)

**Comment by lukeyeager** [comment]
Woah, weird. You snapshotted just fine after iteration 7716, but then it failed to snapshot after iteration 8359 (I'm getting that from the caffe_output.log).

That looks like an OS error to me - are you out of space on your harddrive?

**Comment by joyofdata** [comment]
Nope, 3GB was left at that time.

**Comment by lukeyeager** [comment]
Hmm. Training was 43% done - that doesn't seem like a significant milestone. And it's not like you were doing anything strange with your network since it ran fine for almost 2 hours.

Thanks for the detailed bug report! This seems to be pretty clearly an issue with caffe - I'm going to look around in their [isues](https://github.com/BVLC/caffe/issues) and [user group](https://groups.google.com/forum/#!forum/caffe-users) to see if anyone else has had this problem. Hopefully I'll find something and get back to you soon...

**Comment by joyofdata** [comment]
Thanks for the open and agile communication!

**Comment by lukeyeager** [comment]
Well, you're not alone. Let's wait and see if the caffe developers have run into this before.

https://groups.google.com/forum/#!topic/caffe-users/KGF3s1Ztn88

**Comment by lukeyeager** [comment]
I'm closing this since I can't reproduce it. If anyone else runs into this issue, let me know and I'll pass it along to the Caffe developers.

**Comment by hillaryan** [comment]
I'm also running into this error. I don't have any information other than that which has already been provided. It appears to happen randomly.

**Comment by lukeyeager** [comment]
https://github.com/BVLC/caffe/issues/387#issuecomment-44092533

> Or it's simply a path that isn't writeable

Is it possible that you don't have permission to write to the jobs directory? This could happen if you ran DIGITS with sudo permissions at some point, and now you're trying to run it as yourself.

**Comment by djebm2** [comment]
Same problem here. ere is my output:

`Train net output #2: loss2/loss = 0.0688484 (* 0.3 = 0.0206545 loss)
Iteration 300, lr = 0.001
Iteration 303, loss = 0.485267
Train net output #0: loss = 0.384006 (* 1 = 0.384006 loss)
Train net output #1: loss1/loss = 0.125647 (* 0.3 = 0.0376941 loss)
Train net output #2: loss2/loss = 0.211878 (* 0.3 = 0.0635635 loss)
Iteration 303, lr = 0.001
Iteration 306, loss = 0.353804
Train net output #0: loss = 0.298452 (* 1 = 0.298452 loss)
Train net output #1: loss1/loss = 0.0498573 (* 0.3 = 0.0149572 loss)
Train net output #2: loss2/loss = 0.134635 (* 0.3 = 0.0403906 loss)
Iteration 306, lr = 0.001
Iteration 309, loss = 0.111592
Train net output #0: loss = 0.0954259 (* 1 = 0.0954259 loss)
Train net output #1: loss1/loss = 0.0263562 (* 0.3 = 0.00790686 loss)
Train net output #2: loss2/loss = 0.0275187 (* 0.3 = 0.00825562 loss)
Iteration 309, lr = 0.001
Snapshotting to binary proto file snapshot_iter_312.caffemodel
Snapshotting solver state to binary proto file snapshot_iter_312.solverstate
Check failed: proto.SerializeToOstream(&output)
        `

**Comment by rammohan518** [comment]
Hi, I am also having the same error.
Lukeyeader plz provide us with a detailed steps on how to rectify this problem

**Comment by lukeyeager** [comment]
@rammohan518 I have been unable to reproduce the issue, so I can't offer a solution.

For most people, this seems to be a transient issue. If you simply run it again, it may work.

**Comment by ssqNo1** [comment]
I have had this error too. It turns out that I changed the current path using os.chdir() before the snapshot was taken. 
To inspect, you can set the 'snapshot_prefix' to be a unique name without slash, such as 'my_snapshot', then search the snapshot file by its name.

**Comment by mjohn123** [comment]
I have same problem. I used the newest caffe version. Any one can help me to fix it?
```
2017-02-14 10:39:00 [20170214-103859-f49c] [INFO ] Train Caffe Model task started.
2017-02-14 10:39:00 [20170214-103859-f49c] [INFO ] Task subprocess args: "/home/xxx/caffe/build/tools/caffe train --solver=/home/xxx/digits/digits/jobs/20170214-103859-f49c/solver.prototxt --gpu=0 --weights=/home/xxx/digits/examples/semantic-segmentation/fcn_alexnet.caffemodel"
2017-02-14 10:39:02 [20170214-103859-f49c] [ERROR] Train Caffe Model task failed with error code -6
```

**Comment by public1024** [comment]
after searching the Issues on github & google group, I manage to solve it by change my snap_prefix. It's said that if you don't have the permission to write files , you may run into the error. I hope it will work, if anyone else  have this problem.

**Comment by public1024** [comment]
after searching the Issues on github & google group, I manage to solve it by change my snap_prefix. I recommend you to use abs-path as prefix. It's said that if you don't have the permission to write files , you may run into the error. I hope it will work, if anyone else have this problem.
